pipeline/id_minter/src/test/scala/uk/ac/wellcome/platform/idminter/database/IdentifiersDaoTest.scala

```diff
@@ -35,56 +37,124 @@ class IdentifiersDaoTest
 
       withIdentifiersDao {
         case (identifiersDao, _) =>
-          identifiersDao.saveIdentifier(identifier) shouldBe Success(1)
+          val triedLookup = for {
+            _ <- identifiersDao.saveIdentifiers(List(identifier))
+            lookupResult <- identifiersDao.lookupIds(List(sourceIdentifier))
+          } yield (lookupResult)
+
+          triedLookup shouldBe a[Success[_]]
+          triedLookup.get.found shouldBe Map(sourceIdentifier -> identifier)
+          triedLookup.get.notFound shouldBe empty
+      }
+    }
 
-          val triedLookup = identifiersDao.lookupId(
-            sourceIdentifier = sourceIdentifier
-          )
+    it("retrieves multiple ids from the identifiers database") {
+      val identifiersMap = (1 to 3).map { _ =>
+        val sourceIdentifier = createSourceIdentifier
+        val identifier =
+          createSQLIdentifierWith(sourceIdentifier = sourceIdentifier)
+        (sourceIdentifier, identifier)
+      }.toMap
 
-          triedLookup shouldBe Success(Some(identifier))
+      withIdentifiersDao {
+        case (identifiersDao, _) =>
+          val triedLookup = for {
+            _ <- identifiersDao.saveIdentifiers(identifiersMap.values.toList)
+            lookupResult <- identifiersDao.lookupIds(identifiersMap.keys.toList)
+          } yield (lookupResult)
+
+          triedLookup shouldBe a[Success[_]]
+          triedLookup.get.found shouldBe identifiersMap
+          triedLookup.get.notFound shouldBe empty
       }
     }
 
     it(
-      "does not get an identifier if there is no matching SourceSystem and SourceId") {
-      val identifier = createSQLIdentifier
+      "matches correctly the ids to the sourceIdentifier when the sourceIdentifiers have overlapping values") {
+      val sourceIdentifier = createSourceIdentifier
+      val overlappingSourceIdentifier =
+        sourceIdentifier.copy(ontologyType = "Item")
+      val sourceIdentifiers =
+        List(sourceIdentifier, overlappingSourceIdentifier)
+      val identifiers = sourceIdentifiers.map(sourceIdentifier =>
+        createSQLIdentifierWith(sourceIdentifier = sourceIdentifier))
 
       withIdentifiersDao {
         case (identifiersDao, _) =>
-          identifiersDao.saveIdentifier(identifier) shouldBe Success(1)
+          val triedLookup = for {
+            _ <- identifiersDao.saveIdentifiers(identifiers)
+            lookupResult <- identifiersDao.lookupIds(sourceIdentifiers)
+          } yield (lookupResult)
+
+          triedLookup shouldBe a[Success[_]]
+          triedLookup.get.found shouldBe sourceIdentifiers
+            .zip(identifiers)
+            .toMap
+          triedLookup.get.notFound shouldBe empty
+      }
+    }
 
-          val unknownSourceIdentifier = createSourceIdentifierWith(
-            ontologyType = identifier.OntologyType,
-            value = "not_an_existing_value"
-          )
+    it(
+      "retrieves multiple ids and returns the ones it doesn't have in notFound") {
```

Comment:
```suggestion
      "stores the IDs that aren't in the dao in the notFound field") {
```

(still not in love with the name)